muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/avatar.tsx

```diff
@@ -1,20 +1,56 @@
 import * as React from "react";
 import { getUserImageUrl } from "~/util/modifiers";
-import '~/sass/elements/avatar.scss';
+import "~/sass/elements/avatar.scss";
+import { StateType } from "../../reducers/index";
+import { connect, Dispatch } from "react-redux";
+import { i18nType } from "../../reducers/base/i18n";
 
 interface AvatarProps {
-  hasImage: boolean,
-  id: number,
-  firstName: string
+  hasImage: boolean;
+  id: number;
+  firstName: string;
+  userCategory?: number;
+  i18n: i18nType;
+  avatarAriaLabel?: string;
 }
 
-export default function Avatar(props: AvatarProps){
-  let userCategory = props.id > 10 ? props.id % 10 + 1 : props.id;
-  return props.hasImage ? <object className="avatar-container"
-    data={getUserImageUrl(props.id)}
-    type="image/jpeg">
-     <div className={`avatar avatar--category-${userCategory}`}>{props.firstName[0]}</div>
-  </object> : <div className="avatar-container">
-     <div className={`avatar avatar--category-${userCategory}`}>{props.firstName[0]}</div>
-  </div>
+interface AvatarState {}
+
+class Avatar extends React.Component<AvatarProps, AvatarState> {
```

Comment:
This is still a stateless component so it should remain a function?

Reply:
Removed. Left overs when I tried get i18n locales from redux, but it wasn't necessary.